src/structurizr/api/structurizr_client.py

```diff
@@ -192,25 +228,41 @@ def put_workspace(self, workspace: Workspace) -> None:
                 f"Error message: {body.get('message', '')}"
             )
 
-    def lock_workspace(self) -> bool:
+    def _lock_workspace(self) -> Tuple[bool, bool]:
         """
         Lock the Structurizr workspace.
 
         Returns:
-            bool: `True` if the workspace could be locked, `False` otherwise.
-
+            bool: `True` if lock succeeeded
+            bool: `True` if on paid plan
         """
         request = self._client.build_request("PUT", self._lock_url, params=self._params)
         request.headers.update(self._add_headers(request))
         response = self._client.send(request)
-        logger.debug("%r", response.json())
         response.raise_for_status()
+        logger.debug("%r", response.json())
```

Comment:
FYI, the reason I swapped these lines over is that if the response code isn't 200 (e.g. because you gave a bad URL) then you don't get JSON in the response, but the real error gets masked by an exception trying to parse the JSON.

Reply:
I added a unit test that demonstrates that.

Reply:
Great 👍🏼